packages/next/src/client/components/app-router.tsx

```diff
@@ -229,25 +229,15 @@ function Router({
       }),
     [buildId, children, initialCanonicalUrl, initialTree, initialHead]
   )
-  const [
-    {
-      tree,
-      cache,
-      prefetchCache,
-      pushRef,
-      focusAndScrollRef,
-      canonicalUrl,
-      nextUrl,
-    },
-    dispatch,
-    sync,
-  ] = useReducerWithReduxDevtools(reducer, initialState)
+  const [reducerState, dispatch, sync] =
+    useReducerWithReduxDevtools(initialState)
```

Comment:
What's in this initial state that can't be created before render, at the same time that the queue is created?

Reply:
This relies on data that is streamed in from [app-render](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/app-render/app-render.tsx#L359) (such as the initial tree) and I wasn't seeing how we could easily access that data during hydration.